File: spark/javaingestionframework-master/src/main/java/com/nisum/dataingestionframework/dependencies/DataFrameUtils.java
package com.nisum.dataingestionframework.dependencies;

import com.nisum.dataingestionframework.model.ColumnNameTypeLength;
import com.nisum.dataingestionframework.util.Constants;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.apache.spark.sql.functions;

import java.io.Serializable;
import java.util.Iterator;
import java.util.List;

public class DataFrameUtils implements Serializable {

    private static final Logger logger = LoggerFactory.getLogger(DataFrameUtils.class);

    String columnValue;
    DateTimeUtilities dateTimeUtilities = new DateTimeUtilities();

    public Dataset<Row> dataframeAddColumns(Dataset<Row> dataFrame, List<ColumnNameTypeLength> addColumns) {

        Iterator iterator = addColumns.iterator();

        while(iterator.hasNext()) {

            ColumnNameTypeLength columnNameTypeLength =  (ColumnNameTypeLength)iterator.next();


            if (columnNameTypeLength.getColumnValue().length() > 0) {
                logger.info("getting column value from json request");
                columnValue = columnNameTypeLength.getColumnValue();
            }

            if (columnNameTypeLength.getColumnValue().equalsIgnoreCase(Constants.CURRENTDATE)) {
                logger.info("getting current date from json request");

                columnValue = dateTimeUtilities.getCurrentDate().toString();
            }
            if (columnNameTypeLength.getColumnValue().equalsIgnoreCase(Constants.EPOCH)) {
                logger.info("getting epoch of current time from json request");

                columnValue = dateTimeUtilities.getEpoch().toString();
            }

            dataFrame = dataFrame.withColumn(
                    columnNameTypeLength.getColumnName(),
                    functions.lit(columnValue)
            );


        }
        return dataFrame;
    }

}

File: spark/python_code/ingestionframework-master/src/dependencies/dataframe_utils.py
# -*- coding: utf-8 -*-
"""
Created on 04/12/2020 2:01 PM

@author : Nirav Langaliya
"""
from typing import List

from pyspark.sql import DataFrame
from pyspark.sql import functions as F
from pyspark.sql.types import StringType
from config.config_objects import ColumnNameTypeLength
from dependencies.schema_utilties import data_type_dict
from dependencies.datetime_utilities import *

def dataframe_add_columns(
        dataframe:DataFrame,
        addColumns: List[ColumnNameTypeLength]
):

    for new_column in addColumns:

        if new_column.columnValue:
            column_value = new_column.columnValue

        if new_column.columnName not in dataframe.columns:
            if new_column.columnValue == "current_date":
                column_value = get_current_date()
            if new_column.columnValue == "epoch":
                column_value = get_epoch()

            dataframe = dataframe.withColumn(
                new_column.columnName,
                F.lit(column_value).cast(
                    data_type_dict.get(new_column.columnType, StringType)
                )
            )
        else:
            dataframe = dataframe.withColumn(
                new_column.columnName,
                dataframe[new_column.columnName].cast(
                    data_type_dict.get(new_column.columnType, StringType)
                )
            )

    return dataframe
